### PR TITLE
fix(table): avoid duplicate onOpenChange(false) when closing filter dropdown (Closes #58988)

### LINKED ISSUE #58988: Table 关闭内置筛选面板时会重复触发 onOpenChange(false)
### Reproduction link

[https://codesandbox.io/p/sandbox/table-guan-bi-nei-zhi-shai-xuan-mian-ban-shi-hui-chong-fu-chu-fa-onopenchange-false-3944q6?file=%2Fdemo.tsx](https://codesandbox.io/p/sandbox/table-guan-bi-nei-zhi-shai-xuan-mian-ban-shi-hui-chong-fu-chu-fa-onopenchange-false-3944q6?file=%2Fdemo.tsx)

### Steps to reproduce

1. 打开在线最小复现和浏览器控制台
2. 点击 Status 列的筛选图标打开面板
3. 再次点击筛选图标关闭面板
4. 查看控制台中的 onOpenChange 记录

### What is expected?

一次打开和一次关闭依次输出 true、false。

### What is actually happening?

一次打开和一次关闭依次输出 true、false、false，关闭回调重复执行。

| Environment | Info |
| --- | --- |
| antd | 6.5.3 |
| React | 19.2.8 |
| System | darwin 25.5.0 |
| Browser | - |

---
Table 列同时使用内置 filters 和 filterDropdownProps.onOpenChange 时，关闭筛选面板会连续两次触发 onOpenChange(false)。重复回调会导致调用方的状态同步、埋点或副作用逻辑执行两次。

验证环境：Chrome / Chromium。

---
<sub>Generated by [@ant-design/cli](https://github.com/ant-design/ant-design-cli)</sub>

### PR DESCRIPTION
**Pull request by waterWang**
## Summary

Fixes #58988 — when a Table column uses both built-in filters and `filterDropdownProps.onOpenChange`, closing the filter panel triggers `onOpenChange(false)` twice, causing side effects (state sync, analytics, etc.) to run twice.

### Root cause

In `FilterDropdown.tsx`, the `onVisibleChange` handler calls:

```ts
triggerVisible(newVisible);   // 1st onOpenChange(false)

if (!newVisible && !column.filterDropdown && filterOnClose) {
  onConfirm();                // onConfirm internally calls triggerVisible(false) again
}
```

`onConfirm()` was unconditionally calling `triggerVisible(false)`, so when invoked from `onVisibleChange` after the dropdown already closed, it produced a second `onOpenChange(false)`.

### Fix

Give `onConfirm` a `shouldClose` parameter (default `true` so the "Confirm" button behavior is unchanged). When called from `onVisibleChange` while the panel is already closing, pass `false` to skip the redundant close:

```ts
const onConfirm = (shouldClose = true) => {
  if (shouldClose) {
    triggerVisible(false);
  }
  internalTriggerFilter(getFilteredKeysSync());
};

// onVisibleChange:
if (!newVisible && !column.filterDropdown && filterOnClose) {
  onConfirm(false); // already closing, skip redundant triggerVisible
}
```

### Testing

- [x] `filterOnClose` path: opening then closing the panel now emits exactly `true, false` instead of `true, false, false`
- [x] Confirm button still calls `triggerVisible(false)` (default `shouldClose = true`)
- [x] No behavior change for other call paths (`onReset`, `doFilter`)

Verified against the reproduction from the issue (antd 6.5.3 / React 19.2.8).

### CODE REVIEW

File: components/table/hooks/useFilter/FilterDropdown.tsx [hunk 1]
```diff
@@ -290,8 +290,10 @@ const FilterDropdown = <RecordType extends AnyObject = AnyObject>(
...
   };
 
-  const onConfirm = () => {
-    triggerVisible(false);
+  const onConfirm = (shouldClose = true) => {
+    if (shouldClose) {
+      triggerVisible(false);
+    }
     internalTriggerFilter(getFilteredKeysSync());
   };
```
[the remark this answers is not in the document]
Comment (QDyanbing) [reply components/table/hooks/useFilter/FilterDropdown.tsx:298]: It looks like this regression is not covered by a test yet.

**Comment by QDyanbing** [comment]
> Could you please:
>
> 1. Update the PR description using the repository’s PR template.
> 2. Add a regression test covering the duplicate `onOpenChange(false)` callback.
> 3. Address the CI/type-check issue reported above.
>
> Thanks!

**Comment by QDyanbing** [comment]
你好，想确认一下这个问题目前是否仍在处理中？如果没有，我们会将它分配给其他贡献者处理。谢谢！

Hi, just checking whether you’re still working on this issue. If not, we’ll assign it to another contributor. Thanks!

**Comment by QDyanbing** [comment]
Closing as #59023 has been merged. Thanks!
